docs/deployment/terraform-best-practices.md

```diff
@@ -0,0 +1,196 @@
+delete marker by replacing `BUCKET` and `KEY` as appropriate for the state file
+in question, in the following command:
+
+```bash
+aws s3api list-object-versions \
+  --bucket BUCKET \
+  --prefix KEY \
+  --query "DeleteMarkers[?IsLatest].VersionId | [0]"
+```
+
+If the output from this command is `null`, then there is no delete marker, and
+you may want to double-check your bucket and key values. If the bucket and key
+values are correct, then your state file is either _not_ marked as deleted or
+does not exist at all.
+
+Otherwise, you may remove the delete marker so that the state file no longer
+appears deleted. This will restore the previous version of the file and make it
+the latest version. Run the following command, using the same values for
+`BUCKET` and `KEY` as used in the previous command, and replacing `VERSION_ID`
+with the value output from the previous command:
+
+```bash
+aws s3api delete-object \
+  --bucket BUCKET \
+  --key KEY \
+  --version-id VERSION_ID
+```
+
+### Deny DeleteBucket Action
+
+As an additional measure to protect your Terraform state files from accidental
+loss, it is also recommended that you deny all users the ability to delete the
+bucket itself. At a later time, you may remove this protection when you are
+sure you want to delete the bucket.
+
+To perform this action this via the **AWS Management Console**:
+
+1. Go to the S3 service
+2. Go to the bucket used for storing state files
+3. Click the **Permissions** tab
+4. Click **Bucket Policy**
+5. Add the following policy statement to _deny_ the s3:DeleteBucket action for
+   all (`"*"`) principals, replacing `BUCKET_NAME` with the name of the bucket:
+
+   ```json
+   {
+     "Statement": [
+       {
+         "Sid": "DenyDeleteBucket",
+         "Effect": "Deny",
+         "Principal": "*",
+         "Action": "s3:DeleteBucket",
+         "Resource": "arn:aws:s3:::BUCKET_NAME"
+       }
+     ]
+   }
+   ```
+
+6. Click **Save**
+
+To perform this action via the **AWS CLI** instead, save the JSON shown above
+to a file named `policy.json` and run the following command from the directory
+in which you saved `policy.json`, replacing `BUCKET_NAME` with the name of the
+bucket:
+
+```bash
+aws s3api put-bucket-policy --bucket BUCKET_NAME --policy file://policy.json
+```
+
+Afterwards, remove the `policy.json` file.
+
+## Change Resources Only via Terraform
+
+**All resource changes must be made via Terraform**, otherwise you risk that
+your Terraform state file does not correctly represent the state of your
+deployment resources. Specifically, this means:
+
+- **DO NOT** change deployment resources via the AWS Management Console
+- **DO NOT** change deployment resources via the AWS CLI
+- **DO NOT** change deployment resources via any of the AWS SDKs
+
+Instead, change deployment resources **only** via changes to your Terraform
+files, along with subsequent Terraform commands.
```

Comment:
This is where I need help. I don't know what to add here.

Reply:
Same with the remaining sections. I don't know enough to know what to fill in.

Reply:
@laurenfrederick it was suggested by @yjpa7145 that perhaps we leave these details for another day, particularly since we don't have enough information to fill in details yet (or do we?).

If you can at least supply me with a direction to explore, I'd be happy to experiment on my orphaned resources on an old MAAP deployment I have, which might give us enough information for a future ticket to add more details here.

Reply:
I don't know. Did we talk about this on ix-dev?

Reply:
This feels like a scoping problem.   We should be able to provide a couple of examples of what that looks like for, say, a lambda (terraform will just redeploy it)  but that's a different case than doing something like deleting a security group, and that's different than something in EC2 etc. 

In most cases I've seen, if you lose the state file the advice centers around recovery and in some cases manual import/manipulation of the state file to try to get it to match, then using tags/etc to clean up on AWS.

Could we provide some reading links on mainpulating TF state, tools to find tagged resources (e.g. using something like `aws resourcegroupstaggingapi get-resources --tag-filters Key=Deployment,Values=DEPLOYMENT` -- note this only pulls in roughly 19x resources) and/or document somehow how to find the rest of the resources, but leave the *particulars* out of it?